homeassistant/components/media_player/__init__.py

```diff
@@ -1158,6 +1180,32 @@ async def async_browse_media(
         """
         raise NotImplementedError
 
+    async def async_internal_search_media(
+        self,
+        query: str,
+        media_content_type: MediaType | str | None = None,
+        media_content_id: str | None = None,
+        target_media_classes: set[MediaClass] | None = None,
+    ) -> SearchMedia:
+        return await self.async_search_media(
+            query=SearchMediaQuery(
+                query=query,
+                media_content_type=media_content_type,
+                media_content_id=media_content_id,
+                target_media_classes=target_media_classes,
+            )
+        )
+
+    async def async_search_media(
+        self,
+        query: SearchMediaQuery,
+    ) -> SearchMedia:
+        """Search the media player.
+
+        "media_player/search_media" websocket command.
+        """
```

Comment:
There is also an action.

```suggestion
        """Search the media player."""
```